### ISSUE TITLE
Position-only arguments enforced even on Python versions < 3.8

### ISSUE THREAD

**Issue by matejcik** [microsoft/pyright#2669]
Introduced in 1.1.192, presumably with "Bug Fix: Added missing check in function type compatibility checks for the case where the source type contains position-only parameters but the destination type does not."

The problem does not appear on 1.1.191.

This is a minimal reproducer:
```python
from typing_extensions import Protocol
from io import BytesIO

class Writer(Protocol):
    def write(self, buf: bytes) -> int:
        ...

def foo(w: "Writer") -> None:
    w.write(b"hello")

buf = BytesIO()
foo(buf)
```

pyright version **1.1.193** reports the following:
```
% pyright test.py --pythonversion=3.6                                                                                                    1 ↵
No configuration file found.
No pyproject.toml file found.
stubPath /home/matejcik/a/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/home/matejcik/a/test.py
  /home/matejcik/a/test.py:12:5 - error: Argument of type "BytesIO" cannot be assigned to parameter "w" of type "Writer" in function "foo"
    "BytesIO" is incompatible with protocol "Writer"
      "write" is an incompatible type
        Type "(__buffer: ReadableBuffer) -> int" cannot be assigned to type "(buf: bytes) -> int"
          Position-only parameter mismatch; expected 1 but received 0 (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 0.816sec
```

The complaint is that `write(self, buf: bytes)` is incompatible with BytesIO variant `write(self, buf: ReadableBuffer, /)`.

(which, btw, is NOT indicated by the error message.)

However, I'm in Python 3.6, where position-only arguments don't exist. And indeed, adding the `, /` to the protocol will yield a different error:
```
  /home/matejcik/a/test.py:5:33 - error: Position-only argument separator requires Python 3.8 or newer
```

**Comment by erictraut** [comment]
I think that pyright is correct to emit an error here. The problem is that the protocol allows for keyword arguments, and that will fail at runtime. For example, what if `foo` calls `write` as follows? This is allowed by the protocol but is not allowed by `BytesIO`.

```python
def foo(w: Writer) -> None:
    w.write(buf=b"hello")
```

For versions of Python prior to 3.8, there is no syntactic way of specifying position-only parameters, but there is a convention that pyright and other type checkers support. You simply name the parameters such that they begin with a double underscore. So, if you change the name of parameter `buf` to `__buf`, the error goes away.
